rotkehlchen/chain/ethereum/modules/uniswap/uniswap.py

```diff
@@ -505,6 +531,36 @@ def get_balances(
 
         return protocol_balance.address_balances
 
+    def get_v3_balances(
+        self,
+        addresses: List[ChecksumEthAddress],
+    ) -> AddressToV3LPBalances:
+        """Get the addresses' balances in the Uniswap V3 protocol
+
+        Premium users can request balances either via the Uniswap subgraph or
+        on-chain.
+        """
+        protocol_balance = self.get_v3_balances_chain(addresses)
+
+        known_assets = protocol_balance.known_assets
+        unknown_assets = protocol_balance.unknown_assets
+
+        known_asset_price = self._get_known_asset_price(
+            known_assets=known_assets,
+            unknown_assets=unknown_assets,
+        )
+
+        unknown_asset_price: AssetToPrice = {}
+        if self.premium:
+            unknown_asset_price = self._get_unknown_asset_price_graph(unknown_assets=unknown_assets)  # noqa:E501
```

Comment:
This makes use of the subgraph to query prices. I would guess that you have executed this part of the code and verified that the prices are correct but since we already have a uniswap v3 oracle maybe we can get the price from there instead of using the subgraph?

Reply:
I don't seem to get what you mean here